### ISSUE TITLE
date shows in grid is different from value in database.

### ISSUE THREAD

**Issue by qjcjerry** [nocodb/nocodb#483]
**Please enter the following details**
please provide the following details :
NocoDB used as docker : false
NocoDB version : 
Database used in NC_DB URL  : mysql
Project was created  by clicking : New Project
Database on which spreadsheet is created : mysql 
OS on which NocoDB is running : macos
Node.js version if running as node : 14.17.4
Database version : 5.7.9

**Steps To Reproduce**
1. create a table;
2. add acolumn as date type;
3. add a row, select a date and save;
4. select data in database;

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/6712585/130339158-0dc8e1b2-7b2f-448b-a779-6914310d01db.png)

**Comment by o1lab** [comment]
Tried the same scenario, was unable to reproduce the scenario.

**Comment by pranavxc** [comment]
Please share the data type and timezone details.

**Comment by qjcjerry** [comment]
![image](https://user-images.githubusercontent.com/6712585/130387284-5c0a9a2b-579e-4167-ace8-ca2617c32be9.png)

**Comment by qjcjerry** [comment]
> Please share the data type(database) and timezone details.

![image](https://user-images.githubusercontent.com/6712585/130389542-3e0f8637-7bcb-4ec0-a59b-80c5b6e964ec.png)

**Comment by pranavxc** [comment]
Fixed in `0.11.24`

**Comment by qjcjerry** [comment]
When will this problem be solved. thanks a lot.

**Comment by qjcjerry** [comment]
> When will this problem be solved. thanks a lot.

@o1lab @pranavxc

**Comment by o1lab** [comment]
This is slightly tricky since we have to use database schema's datetime zone. And it will be different for each database. Then in UI, we've to show a time relevant to user browser time.

**Comment by o1lab** [comment]
#492

**Comment by victorshx** [comment]
> This is slightly tricky since we have to use database schema's datetime zone. And it will be different for each database. Then in UI, we've to show a time relevant to user browser time.

Converting the DateTime record to the timezone in the user's browser is not a good idea because the record needs to be in a standard timezone(UTC, etc.). 

I propose this: convert DateTime record according to env(presumably `nc_timezone='Asia/Singapore'` or `nc_timezone=+8` or database timezone.

**Comment by victorshx** [comment]
@o1lab

**Comment by o1lab** [comment]
Right way to go about it is to query timezone database is using and convert it in either directions.

**Comment by o1lab** [comment]
Please try on 0.83.6+ version

**Comment by victorshx** [comment]
> Please try on 0.83.6+ version

Thanks, the correct date is showing now.

**Comment by wingkwong** [comment]
Reopening. The data in DB is correct. However, the values are not correct in NocoDB. Even there is no values in DB, it shows some arbitrary values in NocoDB. 

![image](https://user-images.githubusercontent.com/35857179/146506861-32a398a3-1c61-4c85-9dde-ad7288166f7a.png)

![image](https://user-images.githubusercontent.com/35857179/146506876-6f02da1d-70f8-437f-b7ac-a89375149cb7.png)


Meta:

```json
{
	"validate": {
		"func": [],
		"args": [],
		"msg": []
	},
	"cn": "StartTime",
	"_cn": "StartTime",
	"type": "timestamp",
	"dt": "timestamp",
	"uidt": "DateTime",
	"uip": "",
	"uicn": "",
	"dtx": "specificType",
	"ct": "integer(11)",
	"nrqd": true,
	"rqd": false,
	"ck": false,
	"pk": false,
	"un": false,
	"ai": false,
	"cdf": null,
	"clen": 45,
	"np": null,
	"ns": null,
	"dtxp": "",
	"dtxs": " ",
	"cno": "StartTime",
	"tn": "foo",
	"alias": "StartTime"
}, 
{
	"validate": {
		"func": [],
		"args": [],
		"msg": []
	},
	"cn": "EndTime",
	"_cn": "EndTime",
	"type": "timestamp",
	"dt": "timestamp",
	"uidt": "DateTime",
	"uip": "",
	"uicn": "",
	"dtx": "specificType",
	"ct": "integer(11)",
	"nrqd": true,
	"rqd": false,
	"ck": false,
	"pk": false,
	"un": false,
	"ai": false,
	"cdf": null,
	"clen": 45,
	"np": null,
	"ns": null,
	"dtxp": "",
	"dtxs": " ",
	"cno": "EndTime",
	"tn": "foo",
	"alias": "EndTime"
}
```

DB:

![image](https://user-images.githubusercontent.com/35857179/146505331-abd5aaee-bfa3-470a-a254-e1215968539d.png)

Node: **v12.22.6**
Arch: **x64**
Platform: **linux**
Docker: **false**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **pg**
PackageVersion: **0.83.8**

**Comment by o1lab** [comment]
@wingkwong : please paste the project info and DB timezone

**Comment by wingkwong** [comment]
@o1lab 

**Project info:**

Node: v12.22.6
Arch: x64
Platform: linux
Docker: false
Database: pg
ProjectOnRootDB: false
RootDB: pg
PackageVersion: 0.83.8

**DB Timezone:**
![image](https://user-images.githubusercontent.com/35857179/146517420-e4f2129b-e149-4456-b432-efd737af7667.png)

**Comment by letianwei** [comment]
hello team, any update on this? thanks a lot!!

**Comment by letianwei** [comment]
create new table with datetime field, select the correct  date time as below
![image](https://user-images.githubusercontent.com/39935412/146729346-4e6699cb-77c8-43b2-ab74-54512853a78b.png)

![image](https://user-images.githubusercontent.com/39935412/146729507-6f508cc4-d142-42b3-b801-bf0a3236a34d.png)

show correct
![image](https://user-images.githubusercontent.com/39935412/146729548-30fa5dc1-b93c-4017-ae5c-042d5c10c151.png)

show wrong value after saved
![image](https://user-images.githubusercontent.com/39935412/146729598-19c56543-2be6-4c4d-804a-61b42b63d7c2.png)

edit the value of this field, show
![image](https://user-images.githubusercontent.com/39935412/146729748-b488c428-c70b-4246-aff7-c71961eacb9d.png)

**Comment by letianwei** [comment]
thanks @pranavxc may i know if there has any plan to provide the new release version to include this bug fix?  thanks.

**Comment by o1lab** [comment]
Will be available in next release @letianwei

**Comment by pranavxc** [comment]
Fixed in `0.84.1`
